src/compiler/parser.ts

```diff
@@ -2973,6 +2981,44 @@ namespace ts {
             return parseParenthesizedArrowFunctionExpressionHead(/*allowAmbiguity*/ false);
         }
 
+        function tryParseUnparenthesizedAsyncArrowFunctionExpressionWorker(): ArrowFunction {
+            const isUnParenthesizedAsyncArrowFunction = lookAhead(isUnParenthesizedAsyncArrowFunctionWorker);
+            if (isUnParenthesizedAsyncArrowFunction === Tristate.True) {
+                const node = <ArrowFunction>createNode(SyntaxKind.ArrowFunction);

```

Comment:
can you reuse `parseSimpleArrowFunctionExpression` at this point? It looks almost the same except for `parseModifiersForArrowFunction`, which you don't need because, except for `async`, there are no modifiers on a simple arrow function. (you still need setModifiers, but you can just call it with Async directly)